Framework/Kernel/inc/MantidKernel/SetValueWhenProperty.h

```diff
@@ -0,0 +1,56 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2021 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#pragma once
+
+#include "MantidKernel/IPropertySettings.h"
+#include "MantidKernel/System.h"
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+
+namespace Mantid {
+namespace Kernel {
+
+// Forward declarations
+class IPropertyManager;
+class Property;
+
+class DLLExport SetValueWhenProperty : public IPropertySettings {
```

Comment:
Can we replace `DLLExport` with `MANTID_KERNEL_DLL`? This macro correctly switches on Windows between export/importing symbols based on compiling  vs using the library